src/ProductDetails.js

```diff
@@ -0,0 +1,40 @@
+import useFetch from "./hooks/useFetch";
+import Loading from "./components/Loading"
+import { useParams, useNavigate } from "react-router-dom";
+
+const ProductDetails = (props) => {
+    const {productId} = useParams();
+    const {data, loading, error} = useFetch('products');
+    const product = data[productId - 1];
```

Comment:
Must Fix

You don't want to do this :). You are over-fetching by getting an entire list when all you need is one product.

Just append the productId from your params to the fetch call
```
const {data, loading, error} = useFetch(`products/${productId}`);
```

Reply:
I know this. I knew this. I did this anyway. smh. Anyway, this is the main fix I implemented. Should be all set now.